crates/server/src/store/blocklist/mod.rs

```diff
@@ -0,0 +1,89 @@
+    /// Do not log or block any request when the blocklist is called via consult
+    Disabled,
+    /// Log and block matching requests when the blocklist is called via consult
+    Enforce,
+    /// Log but do not block matching requests when the blocklist is called via consult
+    Log,
+}
+
+/// Configuration for file based zones
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq)]
+#[serde(default, deny_unknown_fields)]
+pub struct BlocklistConfig {
+    /// Support wildcards?  Defaults to true. If set to true, block list entries containing
+    /// asterisks will be expanded to match queries.
+    pub wildcard_match: bool,
+
+    /// Minimum wildcard depth.  Defaults to 2.  Any wildcard entries without at least this many
+    /// static elements will not be expanded (e.g., *.com has a depth of 1; *.example.com has a
+    /// depth of two.) This is meant as a safeguard against an errant block list entry, such as *
+    /// or *.com that might block many more hosts than intended.
+    pub min_wildcard_depth: u8,
+
+    /// Block lists to load.  These should be specified as relative (to the server zone directory)
+    /// paths in the config file.
+    pub lists: Vec<String>,
+
+    /// IPv4 sinkhole IP. This is the IP that is returned when a blocklist entry is matched for an
+    /// A query.  It defaults to 0.0.0.0.
+    pub sinkhole_ipv4: String,
+
+    /// IPv6 sinkhole IP.  This is the IP that is returned when a blocklist entry is matched for a
+    /// AAAA query.  It defaults to ::0
+    pub sinkhole_ipv6: String,
+
+    /// Block TTL. This is the length of time a block response should be stored in the requesting
+    /// resolvers cache, in seconds.  Deafults to 86,400 seconds.
+    pub ttl: u32,
+
+    /// Block message to return to the user.  This is an optional message that, if configured, will
+    /// be returned as a TXT record in the additionals section when a blocklist entry is matched for
+    /// a query.
+    pub block_message: Option<String>,
+
+    /// The consult action controls how the blocklist handles queries where another authority has
+    /// already provided an answer.  By default, it ignores any such queries ("Disabled",) however
+    /// it can be configured to log blocklist matches for those queries ("Log",) or can be
+    /// configured to overwrite the previous responses ("Enforce".)
+    pub consult_action: BlocklistConsultAction,
+}
+
+impl Default for BlocklistConfig {
+    fn default() -> Self {
+        Self {
+            wildcard_match: true,
+            min_wildcard_depth: 2,
+            lists: vec![],
+            sinkhole_ipv4: String::from("0.0.0.0"),
+            sinkhole_ipv6: String::from("::0"),
+            ttl: 86_400,
+            block_message: None,
+            consult_action: BlocklistConsultAction::Disabled,
+        }
+    }
+}
+
+impl BlocklistConfig {
+    /// the set of block lists which should be loaded
+    pub fn get_block_lists(&self) -> &Vec<String> {
```

Comment:
Nit: drop the `get_()` prefix, and make this return a slice: `&[String]`.

Reply:
This function is not needed. I removed it.